packages/compiler-core/src/parse.ts

```diff
@@ -394,6 +397,12 @@ function parseTag(
   advanceBy(context, match[0].length)
   advanceSpaces(context)
 
+  // Component name validation
+  const isComponent = tagType === ElementTypes.COMPONENT
```

Comment:
This error will never happen because only tags starting with n uppercase letter are considered components right now. Instead we should modify the `tagType` logic above so that:

1. If `isNativeTag` option is not provided, only tags starting with uppercase letter are considered components. (This keeps the compiler-core behavior as-is)

2. If `isNativeTag` option *is* provided, anything that is not a native tag is considered a component. In the full build that runs in the browser, we will also need to expose a way to augment the built-in list via `app.config.isNativeTag` (which replaces the current `ignoredElements`, but should support the same signature) so that users can tell Vue to allow known custom elements.

Reply:
`if (/[A-Z-]/.test(tag)) tagType = ElementTypes.COMPONENT `  

Currently, the regular expression above seems to treat tags containing an uppercase letter or `-` as components, not only tags starting with an uppercase letter.

Reply:
I'm going to modify the `tagType` logic above.

Reply:
Maybe the RegExp should be `/\b[A-Z]/`, this matches tags starting with an uppercase letter.

Reply:
Or `/^[A-Z]/`.